multidict/_pair_list.c

```diff
@@ -1115,13 +1115,112 @@ pair_list_update_from_seq(PyObject *op, PyObject *seq)
     return -1;
 }
 
-PyObject *
-_pair_list_calc_identity(PyObject *op, PyObject *key)
+int
+pair_list_eq_to_mapping(PyObject *op, PyObject *other)
 {
     pair_list_t *list = (pair_list_t *)op;
-    return list->calc_identity(key);
+
+    PyObject *identity = NULL;
+    PyObject *ident = NULL;
+
+    PyObject *it = NULL; // iter(other)
+
+    PyObject *items = NULL;
+    PyObject *item = NULL;    
+
+    PyObject *avalue = NULL;
+
+    PyObject *bkey = NULL;
+    PyObject *bvalue = NULL;
+
+    Py_ssize_t pos;
+    
+    int ident_cmp_res;
+    int vals_cmp_res;
+    int is_eq;
+
+    if (pair_list_len(op) != PyMapping_Length(other)) {
+        return 0;
+    }
+
+    // TODO: mb we can iterate by dict insted this overhead
+    items = PyMapping_Items(other);
+    if (items == NULL) {
+        return -1;
+    }
+
+#if PY_MINOR_VERSION <= 5
+    if (PyList_Sort(items) < 0) {
```

Comment:
Let's change iteration order. Roughly
``` for k, v in self:
         if k in other and other[k] == v:
              ...
```
By this, we are independent of a mapping order.
Keep in mind: `other` can be not only `dict` instance but any dict-like object.
The check is a comparison to mapping, not to iterable of pairs.
The dict is not comparable with a list of pairs:
```
In [1]: {'a': 1} == [('a', 1)]
Out[1]: False
```
That's fine for multidict too.

Reply:
Thanks. Good decision!